generator/generator.js

```diff
@@ -2,67 +2,91 @@ const faker = require('faker');
-//writeStyles.write( , 'utf8');
+const writeBag = fs.createWriteStream('bag.csv');
+writeBag.write('userId,itemId,size,quantity,calculatedPrice\n', 'utf8');
 
 
 const generateUser = () => {
-  let user = {
-    email: faker.internet.email(),
-    username: faker.internet.userName()
-  }
-  return user;
+
+  let email = faker.internet.email();
+  let username = faker.internet.userName();
+
+  return `${username},${email}\n`
 };
 
-// const generateProduct = () => {
-//   return faker.commerce.productName();
-// }
-
-// const generateQuantity = () => {
-//   let sizes = {
-//     xs: Math.floor(Math.random() * 10000),
-//     s: Math.floor(Math.random() * 10000),
-//     m: Math.floor(Math.random() * 10000),
-//     l: Math.floor(Math.random() * 10000),
-//     xl: Math.floor(Math.random() * 10000),
-//     xxl: Math.floor(Math.random() * 10000)
-//   }
-//   return sizes;
-// }
-
-// const generateStyles = () => {
-//   let style = {
-//     styleName: faker.commerce.color();
-//   }
-// }
-
-const writeManyEntries = (stream, amount, encoding, cb) => {
+const generateStock = () => {
+  return `${faker.commerce.productName()}\n`;
+}
+
+const generateQuantity = () => {
+  let xs = Math.floor(Math.random() * 10000);
+  let s = Math.floor(Math.random() * 10000);
+  let m = Math.floor(Math.random() * 10000);
+  let l = Math.floor(Math.random() * 10000);
+  let xl = Math.floor(Math.random() * 10000);
+  let xxl = Math.floor(Math.random() * 10000);
+
+  return `${xs},${s},${m},${l},${xl},${xxl}\n`;
+}
+
+const generateStyles = (num) => {
+
+  let imageUrl = "https://loremflickr.com/320/240/clothing";
+  let styleName = faker.commerce.color();
+  let price = faker.commerce.price();
+  let quantityId =  num;
+  return `${imageUrl},${styleName},${price},${quantityId}\n`;
+}
+
+const generateProduct = (num) => {
+  let productId = Math.floor(Math.random() * (1000000 - 1) + 1)
+  let styleId = num;
+
+  return `${productId},${styleId}\n`;
+}
+
+const generateBag = () => {
+
+  let userId = Math.floor(Math.random() * (1000000 - 1) + 1);
+  let itemId = Math.floor(Math.random() * (2000000 - 1) + 1);
+
+  let sizes = ['xs', 's', 'm', 'l', 'xl', 'xxl'];
+  let index = Math.floor(Math.random() * sizes.length);
+  let size = sizes[index];
+
+  let price = faker.commerce.price();
+  let quantity = Math.floor(Math.random() * (15 - 1) + 1)
+
+  return `${userId},${itemId},${size},${quantity},${price}\n`;
+}
+
+
+const writeManyEntries = (stream, amount, dataFunc, encoding, cb) => {
   let i = amount;
   let id = 0;
   const write = () => {
     let ok = true;
     do {
       i -= 1;
       id += 1;
-      const user = generateUser();
-      const data = `${user.username}, ${user.email}\n`;
-      //change to larger divisible
-      if (i % 10 === 0) {
-        console.log('ten more rows written')
+      const data = dataFunc(id);
+
+      if (i % 100000 === 0) {
+        console.log(`a hundred thousand more rows written. ${i} left to go`);
       }
```

Comment:
it seems a '[while](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/do...while)' is needed here to close this loop